tests/test_core.py

```diff
@@ -46,6 +46,25 @@ def test_init_df_with_index(test_pd_df):
     pd.testing.assert_frame_equal(df.timeseries().reset_index(), test_pd_df)
 
 
+def test_init_from_iamdf(test_df_year):
+    # casting an IamDataFrame instance again works
+    df = IamDataFrame(test_df_year)
+
+    # inplace-operations on the new object have effects on the original object
+    df.rename(scenario={'scen_a': 'scen_foo'}, inplace=True)
+    assert all(test_df_year.scenarios().values == ['scen_b', 'scen_foo'])
+
+    # overwrites on the new object do not have effects on the original object
+    df = df.rename(scenario={'scen_foo': 'scen_bar'})
+    assert all(df.scenarios().values == ['scen_b', 'scen_bar'])
+    assert all(test_df_year.scenarios().values == ['scen_b', 'scen_foo'])
+
+
+def test_init_from_iamdf_raises(test_df_year):
+    # casting an IamDataFrame instance again with extra args fails
```

Comment:
should there be any error message in particular here? I can imagine that getting a ValueError just cause you added kwargs would be confusing? Maybe it should say, "If you init with an IamDataFrame, you cannot use kwargs"?

Reply:
added an explicit error message (leaving the door open for valid keyword arguments in the future)